lib/Db/Question.php

```diff
@@ -75,12 +75,7 @@ public function getExtraSettings(): array {
 		return json_decode($this->getExtraSettingsJson() ?: '{}', true); // assoc=true, => Convert to associative Array
 	}
 
-	/**
-	 * @param object|array $extraSettings
-	 */
-	public function setExtraSettings($extraSettings) {
-		// TODO: When the php requirement is >= 8.0 change parameter typing to `object|array` to allow assoc. arrays from `Question::fromParams`
-		
+	public function setExtraSettings(object|array $extraSettings) {
```

Comment:
```suggestion
	public function setExtraSettings(array $extraSettings) {
```

Is it anytime an object? I think this is a leftover from the object -> associative array migration

Reply:
```
{
  "id":87,
  "keyValuePairs":{
    "extraSettings":{
      "shuffleOptions":true,
      "allowOtherAnswer":false
    }
  }
}
```
This is at least what we send from the frontend on updating the extraSettings of a question...

Reply:
Yes but I normally JS objects should be encoded as associative arrays on PHP.

Reply:
@susnux yes, you're right: `Array ( [extraSettings] => Array ( [shuffleOptions] => 1 [allowOtherAnswer] => 1) ) `